model/post.go

```diff
@@ -403,6 +403,166 @@ func (o *Post) Attachments() []*SlackAttachment {
 	return ret
 }
 
+func (o *Post) AttachmentsEqual(input *Post) bool {
+	attachments := o.Attachments()
+	inputAttachments := input.Attachments()
+
+	if len(attachments) != len(inputAttachments) {
+		return false
+	}
+
+	for i := range attachments {
+
+		// Direct comparison of simple types
+
+		if attachments[i].Id != inputAttachments[i].Id {
+			return false
+		}
+
+		if attachments[i].Fallback != inputAttachments[i].Fallback {
+			return false
+		}
+
+		if attachments[i].Color != inputAttachments[i].Color {
+			return false
+		}
+
+		if attachments[i].Pretext != inputAttachments[i].Pretext {
+			return false
+		}
+
+		if attachments[i].AuthorName != inputAttachments[i].AuthorName {
+			return false
+		}
+
+		if attachments[i].AuthorLink != inputAttachments[i].AuthorLink {
+			return false
+		}
+
+		if attachments[i].AuthorIcon != inputAttachments[i].AuthorIcon {
+			return false
+		}
+
+		if attachments[i].Title != inputAttachments[i].Title {
+			return false
+		}
+
+		if attachments[i].TitleLink != inputAttachments[i].TitleLink {
+			return false
+		}
+
+		if attachments[i].Text != inputAttachments[i].Text {
+			return false
+		}
+
+		if attachments[i].ImageURL != inputAttachments[i].ImageURL {
+			return false
+		}
+
+		if attachments[i].ThumbURL != inputAttachments[i].ThumbURL {
+			return false
+		}
+
+		if attachments[i].Footer != inputAttachments[i].Footer {
+			return false
+		}
+
+		if attachments[i].FooterIcon != inputAttachments[i].FooterIcon {
+			return false
+		}
+
+		// Compare length & slice values of fields
+
+		if len(attachments[i].Fields) != len(inputAttachments[i].Fields) {
+			return false
+		}
+
+		for j := range attachments[i].Fields {
```

Comment:
Maybe break these out into equality functions for their various structs? To avoid such a mega function.

Reply:
Split up to each child function in the `Equals` function, makes it indeed nicer to read.